apps/src/code-studio/pd/workshop_dashboard/workshop_index.jsx

```diff
@@ -78,7 +78,8 @@ export class WorkshopIndex extends React.Component {
     const canDelete = this.props.permission.hasAny(
       WorkshopAdmin,
       Organizer,
-      ProgramManager
+      ProgramManager,
+      CsfFacilitator
```

Comment:
I was curious about this permission, so I did some digging.  I think we pass it down if a user is a facilitator for any CSF workshop, not just the current workshop in particular.  In fact, I think this might give a CSF facilitator permission to delete any workshop (on the client).

https://github.com/code-dot-org/code-dot-org/blob/35a0d30e0c0fdfc226ad1ce3f468c49327b0b3ea/dashboard/app/controllers/pd/workshop_dashboard_controller.rb#L21-L23

Reply:
I've left this intact, but also added a `can_delete` property to the serialized workshop returned by the server which checks two additional criteria: that the workshop isn't ended (and hoisted it from client code up to the server); and that the specific user has permission to destroy that workshop.  Only then will the client show the delete button.